src/croniter/croniter.py

```diff
@@ -617,13 +647,14 @@ def _get_prev_nearest(self, x, to_check):
 
     def _get_next_nearest_diff(self, x, to_check, range_val):
         for i, d in enumerate(to_check):
-            if d == "l":
+            if d == "l" and range_val is not None:
                 # if 'l' then it is the last day of month
                 # => its value of range_val
                 d = range_val
             if d >= x:
                 return d - x
-        return to_check[0] - x + range_val
+        if range_val is not None:
+            return to_check[0] - x + range_val
```

Comment:
what are you trying to do/solve/address with these changes toward `range_val` ?
would something not to put in a separate PR ?

Reply:
The year field doesn't have a range.
If current second is 40, and expected second is 10-20, then I can move to next minute.
But if no years match the giving value, then there is no available values anymore, since there is no loop for year field.
So I simply set `range_val` to None to indicate that shouldn't look for next loop.

Reply:
Do you need a further explanation? @kiorky

Reply:
Nope, just didn't had time in the last days, can you please add comments to document year processing tricks ? and i think we will be nearly good.

Reply:
Thanks for your review! XD